### PR TITLE
SDAP-237 Dockerize Collection Manager

### PR DESCRIPTION
**Pull request by eamonford**
https://issues.apache.org/jira/browse/SDAP-237

### CODE REVIEW

File: collection_manager/README.md [hunk 1]
```diff
@@ -1,103 +1,75 @@
...
-    
-    
-
-
-
-
+## Building the Docker image
+From `incubator-sdap-ingester/collection_manager`, run:
 
+    $ docker build . -f docker/Dockerfile -t nexusjpl/collection-manager
```
Comment (tloubrieu-jpl) [inline collection_manager/README.md:75]: The 'containers' subdirectory is missing in the command line.

I have a lot of warning when I run it, it doesn't need for you to change something but I want to make sure you're having the same warnings:
In file included from ext/_yaml.c:596:
ext/_yaml.h:10: warning: "PyString_CheckExact" redefined
 #define PyString_CheckExact PyBytes_CheckExact
 
ext/_yaml.c:486: note: this is the location of the previous definition
   #define PyString_CheckExact          PyUnicode_CheckExact
 
ext/_yaml.c: In function ‘__pyx_pf_5_yaml_get_version_string’:
ext/_yaml.c:1891:17: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   __pyx_v_value = yaml_get_version_string();
                 ^

The docker containers is building on my side.

File: collection_manager/collection_manager/main.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 logging.basicConfig(level=logging.INFO)
 logging.getLogger("pika").setLevel(logging.WARNING)
-logger = logging.getLogger("collection_manager")
+logger = logging.getLogger()
```
Comment (tloubrieu-jpl) [inline collection_manager/collection_manager/main.py:11]: I am usually doing:
logger = logging.getLogger(__name__)

Does it behave the same without the __name__ argument ?

File: collection_manager/collection_manager/main.py [hunk 3]
```diff
@@ -18,34 +18,35 @@ def check_path(path) -> str:
...
                         default='guest',
                         metavar='USERNAME',
                         help='RabbitMQ username. (Default: "guest")')
-    parser.add_argument('--rabbitmq_password',
+    parser.add_argument('--rabbitmq-password',
                         default='guest',
                         metavar='PASSWORD',
                         help='RabbitMQ password. (Default: "guest")')
-    parser.add_argument('--rabbitmq_queue',
+    parser.add_argument('--rabbitmq-queue',
```
Comment (tloubrieu-jpl) [inline collection_manager/collection_manager/main.py:46]: Yes that looks nicer. I think arg parse converts into a proper attribute name on its own.

File: collection_manager/containers/docker/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+python /collection_manager/collection_manager/main.py \
```
Comment (tloubrieu-jpl) [inline collection_manager/containers/docker/entrypoint.sh:3]: The setup.py file can also generate a command line this first line shorted (for example collection-manager):

entry_points={
        'console_scripts': ['collection-manager=collection_manager.collection_manager:main']
    }

File: collection_manager/containers/docker/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  $([[ ! -z "$COLLECTIONS_PATH" ]] && echo --collections-path=$COLLECTIONS_PATH) \
+  $([[ ! -z "$RABBITMQ_HOST" ]] && echo --rabbitmq-host=$RABBITMQ_HOST) \
```
Comment (tloubrieu-jpl) [inline collection_manager/containers/docker/entrypoint.sh:5]: Do we need these lines for the kubernetes/helm integration ?

File: collection_manager/containers/docker/entrypoint.sh (hunk 4 above)
Comment (tloubrieu-jpl) [reply collection_manager/containers/docker/entrypoint.sh:3]: No sure if this is relevant, you will see.

File: collection_manager/collection_manager/main.py (hunk 2 above)
Comment (eamonford) [reply collection_manager/collection_manager/main.py]: good catch -- fixed.

File: collection_manager/README.md (hunk 1 above)
Comment (eamonford) [reply collection_manager/README.md:75]: Thanks! I have now fixed this by moving `/containers/docker` to `/docker` and removing the `kubernetes` directory, as the kubernetes manifest will now reside in the Helm chart. 

I do get those same warnings on my end when building the docker file. Those are coming from compiling the PyYAML library, so they are okay.

File: collection_manager/containers/docker/entrypoint.sh (hunk 4 above)
Comment (eamonford) [reply collection_manager/containers/docker/entrypoint.sh:3]: That's cool, I didn't know about that!

File: collection_manager/containers/docker/entrypoint.sh (hunk 5 above)
Comment (eamonford) [reply collection_manager/containers/docker/entrypoint.sh:5]: These lines will convert environment variables into command line parameters to send to the collection manager script. While it is not strictly necessary, it is a good way of encapsulating the command line syntax of the script and isolating it from the Kubernetes template that launches the docker container.

Comment (tloubrieu-jpl) [reply collection_manager/containers/docker/entrypoint.sh]: Ok I see, that makes sense for a service which runs alone on its docker image.

**Pull request merged**